src/server/job_recov_db.c

```diff
@@ -376,6 +378,46 @@ job_save_db(job *pjob, int updatetype)
 	if (pjob->ji_newjob == 1 && updatetype != SAVEJOB_NEW)
 		return (0);
 
+	if (updatetype == SAVEJOB_NEW && pjob->ji_newjob == 0) {
+		/* If new_job flag is not set but still try to save as a new job, changed to SAVEJOB_FULL*/
+		updatetype = SAVEJOB_FULL;
+		sprintf(log_buffer,"job has already been saved earlier as a new job, thus changing to SAVEJOB_FULL");
+		log_joberr(-1, "job_save_db", log_buffer,  pjob->ji_qs.ji_jobid);
+#ifndef WIN32
+		/* let's print the backtrace to identify the faulty scenario */
+		int total_frames; /* total number of frames returned by backtrace() */
+		int frame_num;
+		void *bt_buffer[BACKTRACE_BUF_SIZE];
+		char **bt_strings;
+
+		/* backtrace() returns current stack addresses */
+		total_frames = backtrace(bt_buffer, BACKTRACE_BUF_SIZE);
```

Comment:
Why are we dumping a backtrace?  There are a lot of error conditions in PBS, but we don't log backtraces.  What makes this one so special?  Also, this is a GNU extension, so we shouldn't use it unless it is correctly guarded.

Reply:
Thanks for the review @bhroam.
Trying to identify a scenario in which PBS tries to save the already saved job again as a new job which causes panic db shutdown. We had identified one and fixed but there might be others so with the help of backtrace will find and fix it without any panic shutdown.
PBS is already using backtrace functionality in qsub command without any special guard and it seems working fine.

Reply:
@Bhagat-Rajput - backtrace in qsub.c is guarded with BACKTRACE_SIZE.

Reply:
@prakashcv13, I didn't noticed that macro BACKTRACE_SIZE is permanently commented and that may be the reason of qsub to work properly.
Do we need to guard the same here ?

Reply:
@Bhagat-Rajput - backtrace() is a good GNU extension that helps in debugging the issue at hand, but this can be added/removed as and when needed. As @bhroam mentioned there are a lot of error conditions, but we do not log backtraces for all of them. It is a debugging tool which a developer can use for debugging, but need not be part of the "committed" code. So, I would suggest removing the backtrace() code.

Reply:
@Bhagat-Rajput i agree with Bhroam that we should guard it. Please use a ifdef for __GNUC__ before using this macro. As to why this is required: The current fix is based on our finding of one specific code flow path that can cause this. There could be more, but it is difficult to find out the flow, and should we hit another flow, this trace will absolutely tell us. Thus this is valuable. We have precedence of adding such things to the code when we had the need to debug or troubleshoot. Recall the famous attributes getting deleted from database after server restart. That code absolutely benefited from dumping data like this.